# Hameed Karim

## Research question/interests

My interest/question is finding out who the worst ever all-star player. This will be done by combing through all the data and analyzing all players who have been all-stars. I will analyze their performance in the all-star games and throughout their careers (regular season and playoffs) ulitizing the statistics given from our data sources.


# MILESTONE 2

In [4]:
import numpy as np

In [5]:
import pandas as pd
path0 = "../data/raw/Game.csv"
pd.read_csv(path0, low_memory = False)

GAME_ID  SEASON_ID  TEAM_ID_HOME TEAM_ABBREVIATION_HOME  \
0      24600001      21946    1610610035                    HUS   
1      24600003      21946    1610610034                    BOM   
2      24600004      21946    1610610025                    CHS   
3      24600002      21946    1610610032                    PRO   
4      24600005      21946    1610610028                    DEF   
...         ...        ...           ...                    ...   
62443  22001076      22020    1610612757                    POR   
62444  22001071      22020    1610612750                    MIN   
62445  22001068      22020    1610612741                    CHI   
62446  22001072      22020    1610612740                    NOP   
62447  22001077      22020    1610612758                    SAC   

                TEAM_NAME_HOME   GAME_DATE MATCHUP_HOME WL_HOME  MIN_HOME  \
0              Toronto Huskies  1946-11-01  HUS vs. NYK       L         0   
1            St. Louis Bombers  1946-11-02  BOM vs. PIT       W         0   
2                Chicago Stags  1946-11-02  CHS vs. NYK       W         0   
3      Providence Steamrollers  1946-11-02  PRO vs. BOS       W         0   
4              Detroit Falcons  1946-11-02  DEF vs. WAS       L         0   
...                        ...         ...          ...     ...       ...   
62443   Portland Trail Blazers  2021-05-16  POR vs. DEN       W       240   
62444   Minnesota Timberwolves  2021-05-16  MIN vs. DAL       W       240   
62445            Chicago Bulls  2021-05-16  CHI vs. MIL       W       240   
62446     New Orleans Pelicans  2021-05-16  NOP vs. LAL       L       240   
62447         Sacramento Kings  2021-05-16  SAC vs. UTA       L       240   

       FGM_HOME  ...  LAST_GAME_VISITOR_TEAM_CITY1  \
0          25.0  ...                           NaN   
1          20.0  ...                           NaN   
2          21.0  ...                           NaN   
3          21.0  ...                           NaN   
4          10.0  ...                           NaN   
...         ...  ...                           ...   
62443      46.0  ...                           DEN   
62444      49.0  ...                           DAL   
62445      45.0  ...                           MIL   
62446      37.0  ...                           LAL   
62447      42.0  ...                           UTA   

       LAST_GAME_VISITOR_TEAM_POINTS  HOME_TEAM_WINS  HOME_TEAM_LOSSES  \
0                                NaN             3.0               3.0   
1                                NaN             5.0               1.0   
2                                NaN             3.0               3.0   
3                                NaN             5.0               1.0   
4                                NaN             1.0               5.0   
...                              ...             ...               ...   
62443                          106.0             1.0               2.0   
62444                          128.0             1.0               2.0   
62445                          108.0             1.0               2.0   
62446                          111.0             1.0               2.0   
62447                          154.0             0.0               3.0   

       SERIES_LEADER  VIDEO_AVAILABLE_FLAG  PT_AVAILABLE  PT_XYZ_AVAILABLE  \
0               Tied                   0.0           0.0               0.0   
1          St. Louis                   0.0           0.0               0.0   
2               Tied                   0.0           0.0               0.0   
3         Providence                   0.0           0.0               0.0   
4         Washington                   0.0           0.0               0.0   
...              ...                   ...           ...               ...   
62443         Denver                   1.0           0.0               0.0   
62444         Dallas                   1.0           0.0               0.0   
62445      Milwaukee                  

In [6]:
path1 = "../data/raw/Player.csv"
pd.read_csv(path1)

id            full_name first_name     last_name  is_active
0       76001       Alaa Abdelnaby       Alaa     Abdelnaby          0
1       76002      Zaid Abdul-Aziz       Zaid    Abdul-Aziz          0
2       76003  Kareem Abdul-Jabbar     Kareem  Abdul-Jabbar          0
3          51   Mahmoud Abdul-Rauf    Mahmoud    Abdul-Rauf          0
4        1505    Tariq Abdul-Wahad      Tariq   Abdul-Wahad          0
...       ...                  ...        ...           ...        ...
4496  1627790           Ante Zizic       Ante         Zizic          1
4497    78647             Jim Zoet        Jim          Zoet          0
4498    78648            Bill Zopf       Bill          Zopf          0
4499  1627826          Ivica Zubac      Ivica         Zubac          1
4500    78650           Matt Zunic       Matt         Zunic          0

[4501 rows x 5 columns]

In [7]:
path2 = "../data/raw/Player_Attributes.csv"
pd.read_csv(path2)

ID FIRST_NAME     LAST_NAME   DISPLAY_FIRST_LAST  \
0       76001       Alaa     Abdelnaby       Alaa Abdelnaby   
1       76002       Zaid    Abdul-Aziz      Zaid Abdul-Aziz   
2       76003     Kareem  Abdul-Jabbar  Kareem Abdul-Jabbar   
3          51    Mahmoud    Abdul-Rauf   Mahmoud Abdul-Rauf   
4        1505      Tariq   Abdul-Wahad    Tariq Abdul-Wahad   
...       ...        ...           ...                  ...   
4495  1627790       Ante         Zizic           Ante Zizic   
4496    78647        Jim          Zoet             Jim Zoet   
4497    78648       Bill          Zopf            Bill Zopf   
4498  1627826      Ivica         Zubac          Ivica Zubac   
4499    78650       Matt         Zunic           Matt Zunic   

     DISPLAY_LAST_COMMA_FIRST  DISPLAY_FI_LAST          PLAYER_SLUG  \
0             Abdelnaby, Alaa     A. Abdelnaby       alaa-abdelnaby   
1            Abdul-Aziz, Zaid    Z. Abdul-Aziz      zaid-abdul-aziz   
2        Abdul-Jabbar, Kareem  K. Abdul-Jabbar  kareem-abdul-jabbar   
3         Abdul-Rauf, Mahmoud    M. Abdul-Rauf   mahmoud-abdul-rauf   
4          Abdul-Wahad, Tariq   T. Abdul-Wahad    tariq-abdul-wahad   
...                       ...              ...                  ...   
4495              Zizic, Ante         A. Zizic           ante-zizic   
4496                Zoet, Jim          J. Zoet             jim-zoet   
4497               Zopf, Bill          B. Zopf            bill-zopf   
4498             Zubac, Ivica         I. Zubac          ivica-zubac   
4499              Zunic, Matt         M. Zunic           matt-zunic   

                BIRTHDATE             SCHOOL  COUNTRY  ... NBA_FLAG  \
0     1968-06-24T00:00:00               Duke      USA  ...        Y   
1     1946-04-07T00:00:00         Iowa State      USA  ...        Y   
2     1947-04-16T00:00:00               UCLA      USA  ...        Y   
3     1969-03-09T00:00:00    Louisiana State      USA  ...        Y   
4     1974-11-03T00:00:00     San Jose State   France  ...        Y   
...                   ...                ...      ...  ...      ...   
4495  1997-01-04T00:00:00                     Croatia  ...        Y   
4496  1953-12-30T00:00:00         Kent State      USA  ...        Y   
4497  1948-06-07T00:00:00           Duquesne      USA  ...        Y   
4498  1997-03-18T00:00:00        Mega Basket  Croatia  ...        Y   
4499  1919-12-19T00:00:00  George Washington      USA  ...        Y   

      GAMES_PLAYED_FLAG  DRAFT_YEAR  DRAFT_ROUND DRAFT_NUMBER   PTS  AST  \
0                     Y        1990            1           25   5.7  0.3   
1                     Y        1968            1            5   9.0  1.2   
2                     Y        1969            1            1  24.6  3.6   
3                     Y        1990            1            3  14.6  3.5   
4                     Y        1997            1           11   7.8  1.1   
...                 ...         ...          ...          ...   ...  ...   
4495                  Y        2016            1           23   6.0  0.6   
4496                  Y   Undrafted    Undrafted    Undrafted   0.3  0.1   
4497                  Y        1970            2           33   2.2  1.4   
4498                  Y        2016            2           32   8.5  1.1   
4499                  Y        1947         None         None   4.9  0.9   

       REB  ALL_STAR_APPEARANCES    PIE  
0      3.3                   0.0    NaN  
1      8.0                   0.0    NaN  
2     11.2                  18.0    NaN  
3      1.9                   0.0    NaN  
4      3.3                   0.0    NaN  
...    ...                   ...    ...  
4495   3.9                   0.0    NaN  
4496   1.1                   0.0    NaN  
4497   0.9                   0.0    NaN  
4498   7.0                   NaN  0.126  
4499   NaN                   0.0    NaN  

[4500 rows x 37 columns]

In [8]:
path3 = "../data/raw/Player_Salary.csv"
pd.read_csv(path3)

slugSeason            nameTeam         namePlayer    statusPlayer  \
0       2020-21       Atlanta Hawks  Bogdan Bogdanovic  current roster   
1       2021-22       Atlanta Hawks  Bogdan Bogdanovic  current roster   
2       2022-23       Atlanta Hawks  Bogdan Bogdanovic  current roster   
3       2023-24       Atlanta Hawks  Bogdan Bogdanovic  current roster   
4       2020-21       Atlanta Hawks    Brandon Goodwin  current roster   
...         ...                 ...                ...             ...   
1287    2020-21  Washington Wizards  Russell Westbrook  current roster   
1288    2021-22  Washington Wizards  Russell Westbrook  current roster   
1289    2022-23  Washington Wizards  Russell Westbrook  current roster   
1290    2020-21  Washington Wizards      Thomas Bryant  current roster   
1291    2021-22  Washington Wizards      Thomas Bryant  current roster   

      isFinalSeason  isWaived  isOnRoster  isNonGuaranteed  isTeamOption  \
0                 0         0           1                0             0   
1                 0         0           1                0             0   
2                 0         0           1                0             0   
3                 1         0           1                0             0   
4                 0         0           1                0             0   
...             ...       ...         ...              ...           ...   
1287              0         0           1                0             0   
1288              0         0           1                0             0   
1289              1         0           1                0             0   
1290              0         0           1                0             0   
1291              1         0           1                0             0   

      isPlayerOption typeContractDetail       value  
0                  0         Guaranteed  18000000.0  
1                  0         Guaranteed  18000000.0  
2                  0         Guaranteed  18000000.0  
3                  1      Player Option  18000000.0  
4                  0         Guaranteed   1701593.0  
...              ...                ...         ...  
1287               0         Guaranteed  41358814.0  
1288               0         Guaranteed  44211146.0  
1289               1      Player Option  47063478.0  
1290               0         Guaranteed   8333333.0  
1291               0         Guaranteed   8666667.0  

[1292 rows x 12 columns]

# MILESTONE 3

## EXPLORATORY DATA ANALYSIS - 

In [9]:
import pandas as pd
import numpy as np
import seaborn as sns

In [10]:
game = pd.read_csv("../data/raw/Game.csv", low_memory = False)
player = pd.read_csv("../data/raw/Player.csv")
playerinfo = pd.read_csv("../data/raw/Player_Attributes.csv")

print(game.shape)
game.info()

(62448, 149)
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 62448 entries, 0 to 62447
Columns: 149 entries, GAME_ID to HISTORICAL_STATUS
dtypes: float64(99), int64(17), object(33)
memory usage: 71.0+ MB


--> **There are a lot of rows (62,448) for this table. We only want to observe finals series of the top 50 all-stars.**

In [11]:
game.head()

GAME_ID  SEASON_ID  TEAM_ID_HOME TEAM_ABBREVIATION_HOME  \
0  24600001      21946    1610610035                    HUS   
1  24600003      21946    1610610034                    BOM   
2  24600004      21946    1610610025                    CHS   
3  24600002      21946    1610610032                    PRO   
4  24600005      21946    1610610028                    DEF   

            TEAM_NAME_HOME   GAME_DATE MATCHUP_HOME WL_HOME  MIN_HOME  \
0          Toronto Huskies  1946-11-01  HUS vs. NYK       L         0   
1        St. Louis Bombers  1946-11-02  BOM vs. PIT       W         0   
2            Chicago Stags  1946-11-02  CHS vs. NYK       W         0   
3  Providence Steamrollers  1946-11-02  PRO vs. BOS       W         0   
4          Detroit Falcons  1946-11-02  DEF vs. WAS       L         0   

   FGM_HOME  ...  LAST_GAME_VISITOR_TEAM_CITY1  LAST_GAME_VISITOR_TEAM_POINTS  \
0      25.0  ...                           NaN                            NaN   
1      20.0  ...                           NaN                            NaN   
2      21.0  ...                           NaN                            NaN   
3      21.0  ...                           NaN                            NaN   
4      10.0  ...                           NaN                            NaN   

   HOME_TEAM_WINS  HOME_TEAM_LOSSES  SERIES_LEADER  VIDEO_AVAILABLE_FLAG  \
0             3.0               3.0           Tied                   0.0   
1             5.0               1.0      St. Louis                   0.0   
2             3.0               3.0           Tied                   0.0   
3             5.0               1.0     Providence                   0.0   
4             1.0               5.0     Washington                   0.0   

   PT_AVAILABLE  PT_XYZ_AVAILABLE  HUSTLE_STATUS  HISTORICAL_STATUS  
0           0.0               0.0            0.0                1.0  
1           0.0               0.0            0.0                1.0  
2           0.0               0.0            0.0                1.0  
3           0.0               0.0            0.0                1.0  
4           0.0               0.0            0.0                1.0  

[5 rows x 149 columns]

In [12]:
player.columns

Index(['id', 'full_name', 'first_name', 'last_name', 'is_active'], dtype='object')

In [13]:
playerinfo.head()

ID FIRST_NAME     LAST_NAME   DISPLAY_FIRST_LAST  \
0  76001       Alaa     Abdelnaby       Alaa Abdelnaby   
1  76002       Zaid    Abdul-Aziz      Zaid Abdul-Aziz   
2  76003     Kareem  Abdul-Jabbar  Kareem Abdul-Jabbar   
3     51    Mahmoud    Abdul-Rauf   Mahmoud Abdul-Rauf   
4   1505      Tariq   Abdul-Wahad    Tariq Abdul-Wahad   

  DISPLAY_LAST_COMMA_FIRST  DISPLAY_FI_LAST          PLAYER_SLUG  \
0          Abdelnaby, Alaa     A. Abdelnaby       alaa-abdelnaby   
1         Abdul-Aziz, Zaid    Z. Abdul-Aziz      zaid-abdul-aziz   
2     Abdul-Jabbar, Kareem  K. Abdul-Jabbar  kareem-abdul-jabbar   
3      Abdul-Rauf, Mahmoud    M. Abdul-Rauf   mahmoud-abdul-rauf   
4       Abdul-Wahad, Tariq   T. Abdul-Wahad    tariq-abdul-wahad   

             BIRTHDATE           SCHOOL COUNTRY  ... NBA_FLAG  \
0  1968-06-24T00:00:00             Duke     USA  ...        Y   
1  1946-04-07T00:00:00       Iowa State     USA  ...        Y   
2  1947-04-16T00:00:00             UCLA     USA  ...        Y   
3  1969-03-09T00:00:00  Louisiana State     USA  ...        Y   
4  1974-11-03T00:00:00   San Jose State  France  ...        Y   

   GAMES_PLAYED_FLAG  DRAFT_YEAR  DRAFT_ROUND DRAFT_NUMBER   PTS  AST   REB  \
0                  Y        1990            1           25   5.7  0.3   3.3   
1                  Y        1968            1            5   9.0  1.2   8.0   
2                  Y        1969            1            1  24.6  3.6  11.2   
3                  Y        1990            1            3  14.6  3.5   1.9   
4                  Y        1997            1           11   7.8  1.1   3.3   

   ALL_STAR_APPEARANCES PIE  
0                   0.0 NaN  
1                   0.0 NaN  
2                  18.0 NaN  
3                   0.0 NaN  
4                   0.0 NaN  

[5 rows x 37 columns]

In [14]:
playerinfo.columns

Index(['ID', 'FIRST_NAME', 'LAST_NAME', 'DISPLAY_FIRST_LAST',
       'DISPLAY_LAST_COMMA_FIRST', 'DISPLAY_FI_LAST', 'PLAYER_SLUG',
       'BIRTHDATE', 'SCHOOL', 'COUNTRY', 'LAST_AFFILIATION', 'HEIGHT',
       'WEIGHT', 'SEASON_EXP', 'JERSEY', 'POSITION', 'ROSTERSTATUS',
       'GAMES_PLAYED_CURRENT_SEASON_FLAG', 'TEAM_ID', 'TEAM_NAME',
       'TEAM_ABBREVIATION', 'TEAM_CODE', 'TEAM_CITY', 'PLAYERCODE',
       'FROM_YEAR', 'TO_YEAR', 'DLEAGUE_FLAG', 'NBA_FLAG', 'GAMES_PLAYED_FLAG',
       'DRAFT_YEAR', 'DRAFT_ROUND', 'DRAFT_NUMBER', 'PTS', 'AST', 'REB',
       'ALL_STAR_APPEARANCES', 'PIE'],
      dtype='object')

In [15]:
player.columns = player.columns.str.upper()
playerdata = pd.merge(playerinfo, player)

print(playerdata.columns)
playerdata.info()
playerdata.head()


Index(['ID', 'FIRST_NAME', 'LAST_NAME', 'DISPLAY_FIRST_LAST',
       'DISPLAY_LAST_COMMA_FIRST', 'DISPLAY_FI_LAST', 'PLAYER_SLUG',
       'BIRTHDATE', 'SCHOOL', 'COUNTRY', 'LAST_AFFILIATION', 'HEIGHT',
       'WEIGHT', 'SEASON_EXP', 'JERSEY', 'POSITION', 'ROSTERSTATUS',
       'GAMES_PLAYED_CURRENT_SEASON_FLAG', 'TEAM_ID', 'TEAM_NAME',
       'TEAM_ABBREVIATION', 'TEAM_CODE', 'TEAM_CITY', 'PLAYERCODE',
       'FROM_YEAR', 'TO_YEAR', 'DLEAGUE_FLAG', 'NBA_FLAG', 'GAMES_PLAYED_FLAG',
       'DRAFT_YEAR', 'DRAFT_ROUND', 'DRAFT_NUMBER', 'PTS', 'AST', 'REB',
       'ALL_STAR_APPEARANCES', 'PIE', 'FULL_NAME', 'IS_ACTIVE'],
      dtype='object')
<class 'pandas.core.frame.DataFrame'>
Int64Index: 4489 entries, 0 to 4488
Data columns (total 39 columns):
 #   Column                            Non-Null Count  Dtype  
---  ------                            --------------  -----  
 0   ID                                4489 non-null   int64  
 1   FIRST_NAME                        4489 non-null   obj

ID FIRST_NAME     LAST_NAME   DISPLAY_FIRST_LAST  \
0  76001       Alaa     Abdelnaby       Alaa Abdelnaby   
1  76002       Zaid    Abdul-Aziz      Zaid Abdul-Aziz   
2  76003     Kareem  Abdul-Jabbar  Kareem Abdul-Jabbar   
3     51    Mahmoud    Abdul-Rauf   Mahmoud Abdul-Rauf   
4   1505      Tariq   Abdul-Wahad    Tariq Abdul-Wahad   

  DISPLAY_LAST_COMMA_FIRST  DISPLAY_FI_LAST          PLAYER_SLUG  \
0          Abdelnaby, Alaa     A. Abdelnaby       alaa-abdelnaby   
1         Abdul-Aziz, Zaid    Z. Abdul-Aziz      zaid-abdul-aziz   
2     Abdul-Jabbar, Kareem  K. Abdul-Jabbar  kareem-abdul-jabbar   
3      Abdul-Rauf, Mahmoud    M. Abdul-Rauf   mahmoud-abdul-rauf   
4       Abdul-Wahad, Tariq   T. Abdul-Wahad    tariq-abdul-wahad   

             BIRTHDATE           SCHOOL COUNTRY  ... DRAFT_YEAR  DRAFT_ROUND  \
0  1968-06-24T00:00:00             Duke     USA  ...       1990            1   
1  1946-04-07T00:00:00       Iowa State     USA  ...       1968            1   
2  1947-04-16T00:00:00             UCLA     USA  ...       1969            1   
3  1969-03-09T00:00:00  Louisiana State     USA  ...       1990            1   
4  1974-11-03T00:00:00   San Jose State  France  ...       1997            1   

   DRAFT_NUMBER   PTS  AST   REB ALL_STAR_APPEARANCES PIE  \
0            25   5.7  0.3   3.3                  0.0 NaN   
1             5   9.0  1.2   8.0                  0.0 NaN   
2             1  24.6  3.6  11.2                 18.0 NaN   
3             3  14.6  3.5   1.9                  0.0 NaN   
4            11   7.8  1.1   3.3                  0.0 NaN   

             FULL_NAME IS_ACTIVE  
0       Alaa Abdelnaby         0  
1      Zaid Abdul-Aziz         0  
2  Kareem Abdul-Jabbar         0  
3   Mahmoud Abdul-Rauf         0  
4    Tariq Abdul-Wahad         0  

[5 rows x 39 columns]

In [25]:
playerdata.describe(include = 'float64')

HEIGHT       WEIGHT          PTS          AST          REB  \
count  4392.000000  4388.000000  4474.000000  4474.000000  4182.000000   
mean     78.063069   211.100501     6.358181     1.437349     2.981444   
std       3.630557    27.010938     4.923566     1.409641     2.286690   
min      65.000000   133.000000     0.000000     0.000000     0.000000   
25%      75.000000   190.000000     2.800000     0.500000     1.400000   
50%      78.000000   210.000000     5.100000     1.000000     2.400000   
75%      81.000000   230.000000     8.600000     1.900000     4.000000   
max      91.000000   360.000000    31.800000    11.200000    22.900000   

       ALL_STAR_APPEARANCES         PIE  
count           4045.000000  429.000000  
mean               0.342645    0.084002  
std                1.473227    0.088903  
min                0.000000   -1.500000  
25%                0.000000    0.067000  
50%                0.000000    0.089000  
75%                0.000000    0.112000  
max               18.000000    0.211000

In [26]:
game.describe(include = 'float64')

FGM_HOME      FGA_HOME   FG_PCT_HOME     FG3M_HOME     FG3A_HOME  \
count  62430.000000  44586.000000  44548.000000  46791.000000  41527.000000   
mean      39.956816     83.900305      0.467941      5.321152     16.547018   
std        6.799560      9.135863      0.063452      4.225780      9.617828   
min        4.000000      0.000000      0.140000      0.000000      0.000000   
25%       35.000000     78.000000      0.427000      2.000000      9.000000   
50%       40.000000     83.000000      0.467000      5.000000     15.000000   
75%       44.000000     89.000000      0.507000      8.000000     22.000000   
max       72.000000    240.000000      5.125000     28.000000     70.000000   

       FG3_PCT_HOME      FTM_HOME      FTA_HOME   FT_PCT_HOME     OREB_HOME  \
count  41142.000000  62427.000000  59412.000000  59409.000000  41240.000000   
mean       0.346580     20.841479     27.359153      0.754420     12.228952   
std        0.154537      7.182259      8.918174      0.098285      4.440478   
min        0.000000      0.000000      0.000000      0.000000      0.000000   
25%        0.258000     16.000000     21.000000      0.692000      9.000000   
50%        0.348000     20.000000     27.000000      0.760000     12.000000   
75%        0.438000     25.000000     33.000000      0.821000     15.000000   
max        1.000000     61.000000     86.000000      4.167000     37.000000   

       ...  LAST_GAME_HOME_TEAM_POINTS  LAST_GAME_VISITOR_TEAM_ID  \
count  ...                61781.000000               6.178100e+04   
mean   ...                  102.120442               1.610613e+09   
std    ...                   14.450548               3.418717e+02   
min    ...                    0.000000               1.610610e+09   
25%    ...                   93.000000               1.610613e+09   
50%    ...                  102.000000               1.610613e+09   
75%    ...                  112.000000               1.610613e+09   
max    ...                  186.000000               1.610613e+09   

       LAST_GAME_VISITOR_TEAM_POINTS  HOME_TEAM_WINS  HOME_TEAM_LOSSES  \
count                   61781.000000    62379.000000      62379.000000   
mean                      103.561354        2.189375          2.149890   
std                        14.657530        1.737457          1.697547   
min                         0.000000        0.000000          0.000000   
25%                        94.000000        1.000000          1.000000   
50%                       104.000000        2.000000          2.000000   
75%                       113.000000        3.000000          3.000000   
max                       184.000000       12.000000         12.000000   

       VIDEO_AVAILABLE_FLAG  PT_AVAILABLE  PT_XYZ_AVAILABLE  HUSTLE_STATUS  \
count          62379.000000  62379.000000       62379.00000   62379.000000   
mean               0.172750      0.148592           0.12376       0.095192   
std                0.384051      0.355688           0.32931       0.293483   
min                0.000000      0.000000           0.00000       0.000000   
25%                0.000000      0.000000           0.00000       0.000000   
50%                0.000000      0.000000           0.00000       0.000000   
75%                0.000000      0.000000           0.00000       0.000000   
max                2.000000      1.000000           1.00000       1.000000   

       HISTORICAL_STATUS  
count       62379.000000  
mean            0.528752  
std             0.499177  
min             0.000000  
25%             0.000000  
50%             1.000000  
75%             1.000000  
max             1.000000  

[8 rows x 99 columns]

In [20]:
playerdata.value_counts(subset = 'ALL_STAR_APPEARANCES', dropna = False)


ALL_STAR_APPEARANCES
0.0     3677
NaN      444
1.0      121
2.0       64
4.0       42
3.0       33
5.0       23
6.0       19
7.0       19
9.0       11
12.0       8
11.0       7
8.0        6
10.0       6
13.0       4
14.0       2
15.0       2
18.0       1
dtype: int64